packages/x-data-grid/src/components/panel/filterPanel/GridFilterInputValue.tsx

```diff
@@ -37,23 +37,23 @@ function GridFilterInputValue(props: GridTypeFilterInputValueProps) {
   } = props;
 
   const filterTimeout = useTimeout();
-  const [filterValueState, setFilterValueState] = React.useState<string | number | undefined>(
-    sanitizeFilterItemValue(item.value, type),
+  const [filterValueState, setFilterValueState] = React.useState<string | undefined>(
+    sanitizeFilterItemValue(item.value),
   );
   const [applying, setIsApplying] = React.useState(false);
   const id = useId();
   const rootProps = useGridRootProps();
 
   const onFilterChange = React.useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
-      const value = sanitizeFilterItemValue(event.target.value, type);
-      setFilterValueState(value);
+      const value = sanitizeFilterItemValue(event.target.value);
 
+      setFilterValueState(value);
       setIsApplying(true);
       filterTimeout.start(rootProps.filterDebounceMs, () => {
         const newItem = {
           ...item,
-          value,
+          value: type === 'number' && Number(value) ? Number(value) : value,
```

Comment:
It should convert `"0"` to number here, shouldn't it?
```suggestion
          value: type === 'number' && !Number.isNaN(Number(value)) ? Number(value) : value,
```

Reply:
Correct. Thanks.

I have extended the test to cover that as well.